dotnet/year-service/Program.cs

```diff
@@ -1,17 +1,42 @@
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Hosting;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using System.Diagnostics;
 
-namespace year_service
-{
-    public class Program
+var builder = WebApplication.CreateBuilder(args);
+
+const string activitySourceName = "honeycomb.examples.year-service-dotnet";
+ActivitySource activitySource = new(activitySourceName);
+
+builder.Services.AddOpenTelemetryTracing(providerBuilder => providerBuilder
+    .SetResourceBuilder(ResourceBuilder.CreateDefault()
+        .AddService(builder.Configuration.GetValue<string>("Otlp:ServiceName"))
+        .AddEnvironmentVariableDetector()
+    )
+    .AddSource(activitySourceName)
+    .AddAspNetCoreInstrumentation()
+    .AddHttpClientInstrumentation()
+    .AddOtlpExporter(options =>
     {
-        public static void Main(string[] args)
-        {
-            CreateHostBuilder(args).Build().Run();
-        }
+        options.Endpoint = new Uri(builder.Configuration.GetValue<string>("Otlp:Endpoint"));
+        var apiKey = builder.Configuration.GetValue<string>("Otlp:ApiKey");
+        var dataset = builder.Configuration.GetValue<string>("Otlp:Dataset");
+        options.Headers = $"x-honeycomb-team={apiKey},x-honeycomb-dataset={dataset}";
+    }));
+
+var app = builder.Build();
+int[] years =
+{
+    2015, 2016, 2017, 2018, 2019, 2020
+};
+
+app.MapGet("/year", () =>
+{
+    using var activity = activitySource.StartActivity("🗓 get-a-year ✨");

```

Comment:
Do we want to use the activity APIs, or the OTel shim over the top that uses OTel nouns and verbs? I'm cool with either

Reply:
I'll switch it to use the OTEL API -- that's more transferable to other languages